src/java/org/apache/cassandra/service/StorageService.java

```diff
@@ -2811,10 +2817,23 @@ else if (!isMember) // prior to this, the node was not a member
         {
             notifyJoined(endpoint);
         }
-
         PendingRangeCalculatorService.instance.update();
     }
 
+    /**
+     * Retrieves the endpoint state from gossiper and applies proper updates
+     * that we may have ignored when the endpoint was not a member.
+     */
+    private void refreshEndpointState(InetAddressAndPort endpoint)
+    {
+        EndpointState epState = Gossiper.instance.getEndpointStateForEndpoint(endpoint);
+        VersionedValue value;
+        if ((value = epState.getApplicationState(ApplicationState.NET_VERSION)) != null)
+            updateNetVersion(endpoint, value);
+        if ((value = epState.getApplicationState(ApplicationState.INDEX_STATUS)) != null)
```

Comment:
what about RPC state and topology ?

Reply:
I verified with the debugger the topology is already set correctly on initialization. 
As for RPC, this is a newly joining node, it has RPC off at the moment of joining, so no need to switch from false to false. 
Anyway, honestly I don't like this cherry-picking here, but the all the alternatives seem to be more complex (like remembering which values were already sent...). 

Although maybe we could have an enum set remembering which values we ignored.